learn/tooling-guide/cli-tools/openapi.md

```diff
@@ -56,12 +58,29 @@ bal openapi -i <openapi-contract-path> --mode service [(-o|--output) output file
 ```
 
 If you want to generate a Client only, you can set the mode as  `client` in the OpenAPI tool. 
-This client can be used in client applications to call the service defined in the OpenAPI file.
+This client can be used in client applications to call the service defined in the OpenAPI file. Enabling `client` mode you can have boilerplate with tests relevant to client remote functions. 
```

Comment:
`Enabling client mode you can have boilerplate with tests relevant to client remote functions` - any reason why we named it `client`? We did a demo today to a customer. He asked the same question. We thought client means - client connector and there might be another mode called `service` or something. But as per above, client means generating some tests. The name `client` is misleading.

Reply:
Let's address this separately.

Reply:
Yeah, The Ballerina OpenAPI Tool has another mode called  `service`  for generating the services that are related to the given OpenAPI specifications. To identifying the code generation we use the mode option to check whether it is `service` or `client`. :)